### PR TITLE
LPS-104933 Add name property to make the reference unique

### PR DESCRIPTION
**Pull request by marcelabc**

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/PageRenderer.soy [hunk 1]
```diff
@@ -206,7 +206,7 @@
 						'fieldEdited': $_handleFieldEdited,
 						'fieldFocused': $_handleFieldFocused
 						] /}
-						{param ref: 'pageRendererFieldRef' + $field.instanceId + $rowIndex + $columnIndex + $pageIndex /}
+						{param ref: 'pageRendererFieldRef' + $field.instanceId + $field.name + $rowIndex + $columnIndex + $pageIndex /}
```
Comment (oliveiraaraujo) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/PageRenderer.soy:209]: @marcelabc, could you see the possibility of js template literals instead concatenation ?
eg.
``` js
{param ref: `pageRendererFieldRef${$field.instanceId}${$field.name}${$rowIndex}${$columnIndex}${$pageIndex}` /}
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/actions/handleFieldRepeated.es.js [hunk 2]
```diff
@@ -36,6 +40,7 @@ export default (pages, name) => {
 			const indexToAddField = currentIndex + 1;
 			const newField = {
 				...field,
+				instanceId: generateInstanceId,
```
Comment (oliveiraaraujo) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/store/actions/handleFieldRepeated.es.js:43]: hi, @marcelabc , what do you think about changing "generateInstanceId"  from properties to method ?
eg.
``` js
generateInstanceId()
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/repeatable.es.js [hunk 3]
```diff
@@ -16,6 +16,10 @@ const FIELD_NAME_REGEX = /(_\w+_)ddm\$\$(.+)\$(\w+)\$(\d+)\$\$(\w+)/;
 
 const NESTED_FIELD_NAME_REGEX = /(_\w+_)ddm\$\$(.+)\$(\w+)\$(\d+)#(.+)\$(\w+)\$(\d+)\$\$(\w+)/;
 
+export const generateInstanceId = Math.random()
+	.toString(36)
+	.substr(2, 8);
```
Comment (oliveiraaraujo) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/util/repeatable.es.js:21]: hi, @marcelabc , what do you think about changing "generateInstanceId" from a properties to a method ?
eg.
``` js
export const generateInstanceId = () => Math.random()
	.toString(36)
	.substr(2, 8);
```

**Pull request closed without merging**